### PR TITLE
Integration tests via OpsWorks ec2

### LINKED ISSUE #169: Integration testing with Chef Automate via test-kitchen
For Automate we want to do integration tests against. We want to:

- spin up Chef Automate server
- push profiles via audit cookbook upload to automate
- run the profile from automate asset store
- verify inspec is able to execute the profile
- verify the data is available in visibility

### PR DESCRIPTION
**Pull request by alexpop**
fixes #169

### CODE REVIEW

File: test/kitchen-automate/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+## Requirements
+
+ * `test-kitchen` installed with `kitchen-ec2` driver.
+ * ENV variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` in order to create and destroy the test ec2 instance.
+ * Access to the private SSH key(`~/.ssh/kitchen-key-compliance.pem`) used in `.kitchen.yml`
+
+## How is it working?
+
+This cookbook uses the `kitchen-ec2` driver in order to spin up an OpsWorks(Chef Server + Automate) instance in ec2 using a previously snapshotted OpsWorks instance.
```
Comment (chris-rock) [inline test/kitchen-automate/README.md]: Should we add `kitchen-ec2` to our Gemfile? https://github.com/chef-cookbooks/audit/blob/master/Gemfile#L23-L27

Comment (chris-rock) [inline test/kitchen-automate/README.md]: Or are you expecting ChefDK?

File: test/kitchen-automate/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+A `kitchen test` will:
+1. Create the ec2 instance
+2. Converge the node with chef_zero using the `default` recipe
+3. Test the success of the reporting using InSpec resources in `test/integration/default/`. This is verifying that compliance scan reports exist in ElasticSearch(used by Chef Visibility) for both converges.
```
Comment (chris-rock) [inline test/kitchen-automate/README.md]: nit: `ElasticSearch (used by Chef Visibility)` space is missing

File: test/kitchen-automate/recipes/harakiri.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+end
+
+service 'atd' do
+  action [ :enable, :start ]
+end
+
+# Run the "halt" command on the instance 4 hours from now
+# Ensures instances are not left hanging around if integration tests fail
+# To prevent an instance from halting use the `atq` and `atrm ID` commands as root
+execute 'echo "halt" | at now + 4 hours' do
```
Comment (chris-rock) [inline test/kitchen-automate/recipes/harakiri.rb]: 👍 cool trick!

File: test/kitchen-automate/recipes/configure_services.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  command <<-EOH
+    automate-ctl reconfigure
+  EOH
+  action :nothing
+end
+
+
+# Because we snapshot the instance, the initial S3 storage config goes away
+# Disable the S3 store for cookbooks:
+delete_lines "Avoid cookbook storage in S3" do
```
Comment (chris-rock) [inline test/kitchen-automate/recipes/configure_services.rb]: we should add a note, that this is only suitable for test environments, not production environments

File: test/kitchen-automate/README.md (hunk 1 above)
Comment (alexpop) [inline test/kitchen-automate/README.md]: only kitchen + kitchen-ec2 and kitchen-inspec are needed. I'll add it

File: test/kitchen-automate/recipes/configure_services.rb (hunk 4 above)
Comment (alexpop) [inline test/kitchen-automate/recipes/configure_services.rb]: I added a note in test/kitchen-automate/README.md

**Comment by chris-rock** [comment]
Great work @alexpop 💯

**Pull request merged**